### PR TITLE
Add binary ops support for the ConstantSolver

### PR DESCRIPTION
**Pull request by JCapucho**
Only implements between scalars since composites require more work

### CODE REVIEW

File: src/proc/constants.rs [hunk 1]
```diff
@@ -253,6 +261,102 @@ impl<'a> ConstantSolver<'a> {
...
+                ConstantInner::Scalar {
+                    value: right_value, ..
+                },
+            ) if discriminant(left_value) == discriminant(right_value) => {
+                let value = match op {
+                    BinaryOperator::Equal => ScalarValue::Bool(left_value == right_value),
+                    BinaryOperator::NotEqual => ScalarValue::Bool(left_value != right_value),
+
+                    _ => match (left_value, right_value) {
+                        (ScalarValue::Sint(a), ScalarValue::Sint(b)) => match op {
```
Comment (kvark) [inline src/proc/constants.rs:291]: let's just wrap it up, i.e. `=> ScalarValue::Sint(match op {...})`
can be applied to other places too

**Review by kvark — APPROVED** [review]

File: src/proc/constants.rs (hunk 1 above)
Comment (JCapucho) [reply src/proc/constants.rs:291]: This isn't possible because in the comparison operations we return a boolean

Comment (kvark) [reply src/proc/constants.rs:291]: If we are talking about scalars only, the booleans could be moved outside and processed together with `Equal`/`NotEqual`.

**Review by kvark — APPROVED** [review]
Sweet!

**Pull request merged**